services/ui-src/src/routes/Dashboard/Dashboard.test.tsx

```diff
@@ -129,6 +130,16 @@ describe("Test dashboard with admin user", () => {
   });
 });
 
+describe("Test dashboard with help desk user", () => {
+  test("Help desk user cannot delete reports", async () => {
+    mockedUseUser.mockReturnValue(mockHelpDeskUser);
+    await act(async () => {
+      await render(dashboardViewWithReports);
+    });
+    expect(screen.queryByAltText("Delete Program")).toBeNull();
+  });
```

Comment:
Might want to add similar tests to ensure state user, state rep, and approver roles all can't delete it either. And depending on the order in which this PR is merged with respect to [this one](https://github.com/CMSgov/mdct-mcr/pull/3726), we'll want to add similar tests for mobile view since the view of that button is repeated there. Might make sense to extract that logic up higher in the component or something so we're not repeating ourselves too much. @scwambach tagging you for visibility on this since you're working on the related dashboard PR.

Reply:
Added the tests for the other users